packages/react-router5/index.d.ts

```diff
@@ -37,22 +37,23 @@ declare module 'react-router5' {
 
     export const Link: ComponentClass<LinkProps>
 
-    export type InjectedRoute = Partial<{
+    export type InjectedRoute = {
+        router: Router
         route: State
         previousRoute: State
```

Comment:
Is this guaranteed to never be null/undefined?

Reply:
Yes, guaranteed, except when no `router` is in context. But I don't think that it's a good idea to make all of them optional/nullable in that case, because we just can't know what's in context.

Reply:
Removing the `Partial` mapped type off this and not declaring anything as optional guarantees that every property will be non-null. That's not the case with this type definition because both `route` and `previousRoute` can be null, as seen [in this test](https://github.com/router5/router5/blob/ef66c71d4dd502c4fc4338f5fcc3a1f84c6e8c85/packages/react-router5/test/main.js#L57-L67). It would not be proper to favor convenience over correctness.